app/src/main/java/com/example/mysimplecoindeck/di/AppModule.kt

```diff
@@ -14,6 +14,6 @@ object AppModule {
 
     @Singleton
     @Provides
-    @Named("repository")
-    fun provideRepository() = CoinRepository()
-}
\ No newline at end of file
+    fun provideCoinRepository(coinRepositoryImpl: CoinRepositoryImpl): CoinRepository =
```

Comment:
This allows us to have a new `AppModule` for tests that provides a different implementation for the `CoinRepository`. We could do something like this for tests instead: 

```kotlin
@Singleton
@Provides
fun provideCoinRepository(fakeCoinRepository: FakeCoinRepository): CoinRepository =
    fakeCoinRepository
```

and FakeCoinRepository is something like: 

```kotlin
class FakeCoinRepository @Inject constructor() : CoinRepository {
    override suspend fun getCoins() = Response(CoinsData(...))

    override suspend fun getCoin(id: Int) = Response(CoinData(...))
}
```

or any other local fake data instead of accessing the real API.